core/src/main/java/org/apache/iceberg/metrics/CounterResultParser.java

```diff
@@ -88,6 +88,14 @@ static CounterResult fromJson(String counterName, JsonNode json) {
 
     String unit = JsonUtil.getString(UNIT, counter);
     long value = JsonUtil.getLong(VALUE, counter);
-    return CounterResult.of(Unit.fromDisplayName(unit), value);
+    return CounterResult.of(toUnit(unit), value);
+  }
+
+  private static Unit toUnit(String unit) {
```

Comment:
I think I'd leave this file alone and update `fromDisplayName` to do the error handling here. Otherwise we have the logic split across multiple places.

Reply:
sgtm. I'll make sure we do the same for other enums that are affected